client/app/controllers/collection/add.js

```diff
@@ -300,31 +384,72 @@ function check_any(conditions) {
 }
 
 
+// Constructs an array containing the parameters or arguments that are used by an
+// action function. Based on the action function's signature, which is an array of
+// strings that describe how to map the parameters from their keys on the controller
+// to the position in the function's arguments. This function constructs the array
+// that the action function gets applied with
+//
+// Future improvements:
+//
+// - Pass in only the signature, args, and parameters objects to the function;
+//   the function does not require the whole action object.
+
 function cons_arg_arr(action) {
+
     const parameters = this.get('parameters');
+
+    // The signature is an array of keys that descripbe the position of parameters.
     let argarr = action.signature.map((key) => {
+
         // Default to undefined.
         var value = undefined;
+
         // First try to use the parameter.
         if (typeof action.parameters === 'object' &&
+            typeof action.parameters[key] === 'object'
+        ) {
+
+            let exists = false;
+            let pkeys = Object.keys(parameters);
+
+            pkeys.forEach((pkey) => {
+                if (parameters[pkey] === action.parameters[key]) {
+                    value = parameters[pkey];
+                    exists = true;
+                }
+            });
+
+            if (!exists) console.log("No action with that key exists in the controller's action array. client/app/controllers/collection/add.js, line 421]");
```

Comment:
There was a bug here where the parameters weren't matching up right; this check ensures parameters aren't duplicated and the like.